From bogus@does.not.exist.com  Tue Sep  2 20:50:00 2003
From: bogus@does.not.exist.com (Anonymous CVS Login)
Date: Tue, 02 Sep 2003 12:50:00 -0700
Subject: [SDL-CVS] Module SDL12: Change committed
Message-ID: <E19uHA8-00057Q-00@twomix.devolution.com>

This is a multipart message as defined by RFC2049 (MIME)

------=MIME.e9947546717eb6f9bb3c705e84385ccc
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Commiter   : cvs
CVSROOT    : /home/sdlweb/libsdl.org/cvs
Module     : SDL12
Commit time: 2003-09-02 19:50:00 UTC

Log message:

Severe bugfix: format->BitsPerPixel not checked in ListModes function

Modified files:
     src/video/xbios/SDL_xbios.c

------=MIME.e9947546717eb6f9bb3c705e84385ccc
Content-Type: text/plain; name="SDL12.20030902.195000.cvs.diff"
Content-Disposition: attachment; filename="SDL12.20030902.195000.cvs.diff"
Content-Transfer-Encoding: 8bit

Index: SDL12/src/video/xbios/SDL_xbios.c
diff -u SDL12/src/video/xbios/SDL_xbios.c:1.9 SDL12/src/video/xbios/SDL_xbios.c:1.10
--- SDL12/src/video/xbios/SDL_xbios.c:1.9	Wed Feb 12 07:24:39 2003
+++ SDL12/src/video/xbios/SDL_xbios.c	Tue Sep  2 12:49:50 2003
@@ -22,7 +22,7 @@
 
 #ifdef SAVE_RCSID
 static char rcsid =
- "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
+ "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
 #endif
 
 /*
@@ -400,9 +400,19 @@
 
 static SDL_Rect **XBIOS_ListModes(_THIS, SDL_PixelFormat *format, Uint32 flags)
 {
+	unsigned int numlist;
+
 	/* 8 bits -> list 0 */
 	/* 16 bits -> list 1 */
-	return(SDL_modelist[(format->BitsPerPixel)>>4]);
+	if (format->BitsPerPixel == 15) {
+		return NULL;
+	}
+
+	numlist = (format->BitsPerPixel)>>4;
+	if (numlist>1)
+		return NULL;
+
+	return(SDL_modelist[numlist]);
 }
 
 static void XBIOS_FreeBuffers(_THIS)
------=MIME.e9947546717eb6f9bb3c705e84385ccc--



From bogus@does.not.exist.com  Wed Sep  3 21:07:26 2003
From: bogus@does.not.exist.com (Anonymous CVS Login)
Date: Wed, 03 Sep 2003 13:07:26 -0700
Subject: [SDL-CVS] Module SDL12: Change committed
Message-ID: <E19uduY-0000xt-00@twomix.devolution.com>

This is a multipart message as defined by RFC2049 (MIME)

------=MIME.0d9b6d4fb0ac7f37b9ef385d1fcdeb0c
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Commiter   : cvs
CVSROOT    : /home/sdlweb/libsdl.org/cvs
Module     : SDL12
Commit time: 2003-09-03 20:07:26 UTC

Log message:

Finally bugfixed: the simpler the better

Modified files:
     src/video/xbios/SDL_xbios.c

------=MIME.0d9b6d4fb0ac7f37b9ef385d1fcdeb0c
Content-Type: text/plain; name="SDL12.20030903.200726.cvs.diff"
Content-Disposition: attachment; filename="SDL12.20030903.200726.cvs.diff"
Content-Transfer-Encoding: 8bit

Index: SDL12/src/video/xbios/SDL_xbios.c
diff -u SDL12/src/video/xbios/SDL_xbios.c:1.10 SDL12/src/video/xbios/SDL_xbios.c:1.11
--- SDL12/src/video/xbios/SDL_xbios.c:1.10	Tue Sep  2 12:49:50 2003
+++ SDL12/src/video/xbios/SDL_xbios.c	Wed Sep  3 13:07:16 2003
@@ -22,7 +22,7 @@
 
 #ifdef SAVE_RCSID
 static char rcsid =
- "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
+ "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
 #endif
 
 /*
@@ -400,19 +400,13 @@
 
 static SDL_Rect **XBIOS_ListModes(_THIS, SDL_PixelFormat *format, Uint32 flags)
 {
-	unsigned int numlist;
-
 	/* 8 bits -> list 0 */
 	/* 16 bits -> list 1 */
-	if (format->BitsPerPixel == 15) {
+	if ((format->BitsPerPixel != 8) && (format->BitsPerPixel !=16)) {
 		return NULL;
 	}
 
-	numlist = (format->BitsPerPixel)>>4;
-	if (numlist>1)
-		return NULL;
-
-	return(SDL_modelist[numlist]);
+	return(SDL_modelist[(format->BitsPerPixel)>>4]);
 }
 
 static void XBIOS_FreeBuffers(_THIS)
------=MIME.0d9b6d4fb0ac7f37b9ef385d1fcdeb0c--



From bogus@does.not.exist.com  Fri Sep  5 16:18:56 2003
From: bogus@does.not.exist.com (Anonymous CVS Login)
Date: Fri, 05 Sep 2003 08:18:56 -0700
Subject: [SDL-CVS] Module SDL12: Change committed
Message-ID: <E19vIMS-0006rb-00@twomix.devolution.com>

This is a multipart message as defined by RFC2049 (MIME)

------=MIME.25c0ddda396f4479024bea6f2b472503
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Commiter   : cvs
CVSROOT    : /home/sdlweb/libsdl.org/cvs
Module     : SDL12
Commit time: 2003-09-05 15:18:56 UTC

Log message:

Fixed X11 mode line crash with only one video mode (thanks Alan!)

Modified files:
     src/video/x11/SDL_x11modes.c

------=MIME.25c0ddda396f4479024bea6f2b472503
Content-Type: text/plain; name="SDL12.20030905.151856.cvs.diff"
Content-Disposition: attachment; filename="SDL12.20030905.151856.cvs.diff"
Content-Transfer-Encoding: 8bit

Index: SDL12/src/video/x11/SDL_x11modes.c
diff -u SDL12/src/video/x11/SDL_x11modes.c:1.19 SDL12/src/video/x11/SDL_x11modes.c:1.20
--- SDL12/src/video/x11/SDL_x11modes.c:1.19	Sat Jun 28 14:52:26 2003
+++ SDL12/src/video/x11/SDL_x11modes.c	Fri Sep  5 08:18:46 2003
@@ -22,7 +22,7 @@
 
 #ifdef SAVE_RCSID
 static char rcsid =
- "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
+ "@(#) $Id: libsdl-1.2.6-fullscreen.patch,v 1.1 2003/11/01 08:19:15 mr_bones_ Exp $";
 #endif
 
 /* Utilities for getting and setting the X display mode */
@@ -120,7 +120,7 @@
                     goto match;
             }
             qsort(modes, nmodes, sizeof *modes, cmpmodes);
-            for ( i = nmodes-1; i >= 0 ; i-- ) {
+            for ( i = nmodes-1; i > 0 ; i-- ) {
 		if ( ! best_width ) {
                     if ( (modes[i]->hdisplay >= width) &&
                          (modes[i]->vdisplay >= height) ) {
------=MIME.25c0ddda396f4479024bea6f2b472503--

